File: workflow/plugins/python/convert/convert_parse_json/convert_parse_json.py
"""Workflow plugin: parse JSON string."""

import json

from ...base import NodeExecutor


class ConvertParseJson(NodeExecutor):
    """Parse JSON string to object."""

    node_type = "convert.parseJson"
    category = "convert"
    description = "Parse JSON string to object"

    def execute(self, inputs, runtime=None):
        text = inputs.get("text", "")
        try:
            result = json.loads(text)
            return {"result": result}
        except json.JSONDecodeError as e:
            return {"result": None, "error": str(e)}

File: workflow/plugins/go/convert/convert_parse_json/convert_parse_json.go
// Package convert_parse_json provides a workflow plugin for parsing JSON strings.
package convert_parse_json

import (
	"encoding/json"
)

// ConvertParseJson implements the NodeExecutor interface for parsing JSON strings.
type ConvertParseJson struct {
	NodeType    string
	Category    string
	Description string
}

// NewConvertParseJson creates a new ConvertParseJson instance.
func NewConvertParseJson() *ConvertParseJson {
	return &ConvertParseJson{
		NodeType:    "convert.parse_json",
		Category:    "convert",
		Description: "Parse a JSON string to object",
	}
}

// Execute runs the plugin logic.
func (p *ConvertParseJson) Execute(inputs map[string]interface{}, runtime interface{}) map[string]interface{} {
	str, ok := inputs["string"].(string)
	if !ok {
		return map[string]interface{}{"result": nil, "error": "string is required"}
	}

	var result interface{}
	if err := json.Unmarshal([]byte(str), &result); err != nil {
		return map[string]interface{}{"result": nil, "error": err.Error()}
	}

	return map[string]interface{}{"result": result}
}
